api/src/test/java/org/pmiops/workbench/db/dao/UserDaoTest.java

```diff
@@ -508,6 +508,17 @@ public void testFindUserByUsernames() {
     assertThat(users).containsExactly(userWithoutSurvey);
   }
 
+  @Test
+  public void testSatisfyCAPSRequirementToStoreUserAddresses() {
+    // https://docs.google.com/document/d/1jK66dh7qP9VOL3e4GoFATFZMLvgBl81hvvTL6WKklYo
+    DbUser user = new DbUser();
+    DbAddress address = createAddress();
+    user.setAddress(address);
+    user = userDao.save(user);
+
+    assertThat(user.getAddress().getId()).isEqualTo(address.getId());
```

Comment:
We recently wanted to remove the address field altogether. Halfway through the work, we learned about CAPS requirements that require us to collect and store user addresses. To prevent us from regressing on this requirement again, we discussed adding a regression test or a comment.

I think an ideal test would validate that "users must have an address because of these CAPS requirements" but our code doesn't enforce address requirements that at a low level. We have a `@OneToOne` annotation on the relationship between users and addresses but it's still possible to save users without addresses.

We have some really high level tests like "updateProfile_badRequest_nullAddress" that check for this. However, they don't mention the CAPS requirement (why we actually care if it's null) and are very high level (all the way at the API) which doesn't prevent another code path from breaking the requirement.

All this test is doing is adding a couple uses of `setAddress` and `getAddress` to ensure saving addresses works and making it clear why we need addresses - to satisfy a CAPS requirement. Mostly, it's an exercise in "declaring product requirements as automated tests" instead of relying entirely on comments. I'd understand if you'd prefer a comment here.

Open to suggestions!

(I had a couple pending comments I mistakenly didn't submit that may answer this as well)